### ISSUE TITLE
User Profile Permissions Concept

### ISSUE THREAD

**Issue by jterps08** [PowerlineApp/powerline-mobile#554]
Allow the user to define which elements of their user profile they are willing to share with Powerline, Groups, and Users. Some will be un-editable. Others will be editable. These will be respected above the Group Permissions request... So, if a Group asks for the user's Name and the user wants to join that group, the User Profile Permissions take priority and if that field is not shared, it will not be shared and the user cannot join the group.

**Comment by austinpapp** [comment]
From a group pov. 
What if we approach this as the group wants the following info and the user grants per group the info the group wants. I'd see us having far less joins if we allow users to simply disable whatever. 

For the users viewing I think we grant the user the ability to disable other users seeing certain things.

**Comment by jterps08** [comment]
Agreed. We actually already have this concept in place. User is alerted to all information that the Group leader(s) want/ will see if they join the group.
> From a group pov.
> What if we approach this as the group wants the following info and the user grants per group the info the group wants. I'd see us having far less joins if we allow users to simply disable whatever.

I also agree with this so that users can define what general public vs approved followers can see: 
> For the users viewing I think we grant the user the ability to disable other users seeing certain things.

The original issue was created in response to concerns that we are asking for the address during registration. I think this is simply an issue of better communicating why we are asking (to link to electeds/etc) because that information is never exposed to other users or the general public.
